src/dotnet/Core/Uploads/VideoUploadProcessor.cs

```diff
@@ -1,38 +1,71 @@
 using System.Net.Mime;
 using FFMpegCore;
+using FFMpegCore.Enums;
+using FFMpegCore.Pipes;
+using Stl.IO;
 
 namespace ActualChat.Uploads;
 
 public class VideoUploadProcessor(ILogger<VideoUploadProcessor> log) : IUploadProcessor
 {
     private ILogger<VideoUploadProcessor> Log { get; } = log;
 
-    public bool Supports(FileInfo file)
-        => file.ContentType.OrdinalIgnoreCaseContains("video");
+    public bool Supports(string contentType)
+        => MediaTypeExt.IsVideo(contentType);
 
-    public async Task<ProcessedFileInfo> Process(FileInfo file, CancellationToken cancellationToken)
+    public async Task<ProcessedFile> Process(UploadedFile file, CancellationToken cancellationToken)
     {
-        var size = await GetVideoDimensions(file, cancellationToken).ConfigureAwait(false);
-        return size != null
-            ? new ProcessedFileInfo(file, size)
-            : new ProcessedFileInfo(file with { ContentType = MediaTypeNames.Application.Octet, }, null);
+        var (needConversion, size) = await GetVideoInfo(file, cancellationToken).ConfigureAwait(false);
+        if (size == null)
```

Comment:
does size == null mean we failed to read video file info?

Reply:
right, or if video file has no video streams